### PR TITLE
WIP, fork, fix slashing power calc

### PR DESCRIPTION
**Pull request by Kubuxu**
TODO: fix state

### CODE REVIEW

File: chain/actors/actor_storagepower.go [hunk 1]
```diff
@@ -657,6 +659,18 @@ func checkProofSubmissionsAtH(vmctx types.VMContext, self *StoragePowerState, he
 			return aerrors.Escalate(err, "parsing miner address")
 		}
 
+		if vmctx.BlockHeight() >= build.ForkMissingSnowballs {
```
Comment (whyrusleeping) [inline chain/actors/actor_storagepower.go]: just use `>`

File: build/forks.go [hunk 2]
```diff
@@ -5,3 +5,5 @@ const ForkBlizzardHeight = 6288
 const ForkFrigidHeight = 7950
 
 const ForkBootyBayHeight = 11000
+
+const ForkMissingSnowballs = 32000
```
Comment (magik6k) [inline build/forks.go]: Will need bump

File: chain/actors/actor_miner2.go [hunk 3]
```diff
@@ -33,7 +33,10 @@ func (sma StorageMinerActor2) Exports() []interface{} {
 		//8:  sma.DePledge,
 		9:  sma.GetOwner,
 		10: sma.GetWorkerAddr,
-		11: sma.GetPower, // TODO: Remove
+		11: withUpdates(
+			update{0, sma.GetPower},
+			update{build.ForkMissingSnowballs, sma.GetPower2},
+		), // TODO: Remove
```
Comment (magik6k) [inline chain/actors/actor_miner2.go]:
```suggestion
		), // FORK
```

File: chain/actors/actor_miner2.go [hunk 4]
```diff
@@ -404,6 +407,19 @@ func (sma StorageMinerActor2) GetPower(act *types.Actor, vmctx types.VMContext,
 	return self.Power.Bytes(), nil
 }
 
+func (sma StorageMinerActor2) GetPower2(act *types.Actor, vmctx types.VMContext, params *struct{}) ([]byte, ActorError) {
+	_, self, err := loadState(vmctx)
+	if err != nil {
+		return nil, err
+	}
+
+	if self.SlashedAt != 0 {
```
Comment (magik6k) [inline chain/actors/actor_miner2.go]: TODO: check if there's anything that calls this that assumes real power even if slashed (some state api methods do that iirc)

File: chain/actors/actor_storagepower.go (hunk 1 above)
Comment (Kubuxu) [reply chain/actors/actor_storagepower.go]: The `withUpdates` does `>=`.

**Pull request merged**